### PR TITLE
Re-generate for ML 2021-10

### PR DESCRIPTION
**Pull request by HarveyLink**
# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [ ] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/main/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running `generate.ps1`. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version.
	
	**Note: We have recently updated the PSH module called by `generate.ps1` to emit additional data. This would help reduce/eliminate the Code Verification check error. Please run following command**:

	    `dotnet msbuild eng/mgmt.proj /t:Util /p:UtilityName=InstallPsModules`

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

**Review by harneetvirk — APPROVED** [review]

File: sdk/machinelearningservices/Azure.ResourceManager.MachineLearning/src/Azure.ResourceManager.MachineLearning.csproj [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <Version>1.0.0-beta.1</Version>
+    <Description>
+      Azure management client SDK for Azure resource provider MachineLearningServices.
```
Comment (harneetvirk) [inline sdk/machinelearningservices/Azure.ResourceManager.MachineLearning/src/Azure.ResourceManager.MachineLearning.csproj:5]: We should update the description from MachineLearningServices to MachineLearning

File: sdk/machinelearningservices/Azure.ResourceManager.MachineLearning/src/autorest.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+Run `dotnet build /t:GenerateCode` to generate code.
+
+``` yaml
+
+azure-arm: true
+library-name: MachineLearning
+require: https://raw.githubusercontent.com/Azure/azure-rest-api-specs/0bff4d0f259847b1fc97a4ca8f98b8c40d672ba5/specification/machinelearningservices/resource-manager/readme.md
+tag: package-2022-02-01-preview
+modelerfour:
+    lenient-model-deduplication: true
```
Comment (Yao725) [inline sdk/machinelearningservices/Azure.ResourceManager.MachineLearning/src/autorest.md:12]: I think we should remove this configuration.

File: sdk/machinelearningservices/Azure.ResourceManager.MachineLearning/src/autorest.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+clear-output-folder: true
+namespace: Azure.ResourceManager.MachineLearning
+skip-csproj: true
+no-property-type-replacement: 
+- ResourceId
+- VirtualMachineImage
+directive:
```
Comment (Yao725) [inline sdk/machinelearningservices/Azure.ResourceManager.MachineLearning/src/autorest.md:19]: Please add the `rename rule` part here, I saw some acronyms that don't meet our naming convention.

File: sdk/machinelearningservices/Azure.ResourceManager.MachineLearning/src/Generated/Models/AssignedUser.cs [hunk 4]
```diff
@@ -7,7 +7,7 @@
 
 using System;
 
-namespace Azure.ResourceManager.MachineLearningServices.Models
+namespace Azure.ResourceManager.MachineLearning.Models
 {
     /// <summary> A user that can be assigned to a compute instance. </summary>
     public partial class AssignedUser
```
Comment (Yao725) [inline sdk/machinelearningservices/Azure.ResourceManager.MachineLearning/src/Generated/Models/AssignedUser.cs:13]: Are the two properties `Guid`?

File: sdk/machinelearningservices/Azure.ResourceManager.MachineLearning/src/Generated/Models/AutoMLVertical.cs [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+#nullable disable
+
+namespace Azure.ResourceManager.MachineLearning.Models
+{
+    /// <summary>
+    /// AutoML vertical class.
+    /// Base class for AutoML verticals - TableVertical/ImageVertical/NLPVertical
+    /// </summary>
+    public partial class AutoMLVertical
```
Comment (Yao725) [inline sdk/machinelearningservices/Azure.ResourceManager.MachineLearning/src/Generated/Models/AutoMLVertical.cs:14]: `ML` or `Ml`?

File: sdk/machinelearningservices/Azure.ResourceManager.MachineLearning/src/Generated/Models/BatchDeploymentDataPatch.cs [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        {
+            Tags = new ChangeTrackingDictionary<string, string>();
+        }
+
+        /// <summary> Managed service identity (system assigned and/or user assigned identities). </summary>
+        public PartialManagedServiceIdentity Identity { get; set; }
+        /// <summary> Metadata used by portal/tooling/etc to render different UX experiences for resources of the same type. </summary>
+        public string Kind { get; set; }
+        /// <summary> The geo-location where the resource lives. </summary>
+        public string Location { get; set; }
```
Comment (Yao725) [inline sdk/machinelearningservices/Azure.ResourceManager.MachineLearning/src/Generated/Models/BatchDeploymentDataPatch.cs:27]: `AzureLocation`?

File: sdk/machinelearningservices/Azure.ResourceManager.MachineLearning/src/Generated/Models/BillingCurrency.cs [hunk 7]
```diff
@@ -8,14 +8,14 @@
...
     /// <summary> Three lettered code specifying the currency of the VM price. Example: USD. </summary>
     public readonly partial struct BillingCurrency : IEquatable<BillingCurrency>
     {
         private readonly string _value;
 
-        /// <summary> Determines if two <see cref="BillingCurrency"/> values are the same. </summary>
+        /// <summary> Initializes a new instance of <see cref="BillingCurrency"/>. </summary>
         /// <exception cref="ArgumentNullException"> <paramref name="value"/> is null. </exception>
         public BillingCurrency(string value)
         {
```
Comment (Yao725) [inline sdk/machinelearningservices/Azure.ResourceManager.MachineLearning/src/Generated/Models/BillingCurrency.cs:21]: `Usd`?

File: sdk/machinelearningservices/Azure.ResourceManager.MachineLearning/src/Generated/Models/Compute.cs [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+        /// <summary> The provision state of the cluster. Valid values are Unknown, Updating, Provisioning, Succeeded, and Failed. </summary>
+        public ProvisioningState? ProvisioningState { get; }
+        /// <summary> The description of the Machine Learning compute. </summary>
+        public string Description { get; set; }
+        /// <summary> The time at which the compute was created. </summary>
+        public DateTimeOffset? CreatedOn { get; }
+        /// <summary> The time at which the compute was last modified. </summary>
+        public DateTimeOffset? ModifiedOn { get; }
+        /// <summary> ARM resource id of the underlying compute. </summary>
+        public string ResourceId { get; set; }
```
Comment (Yao725) [inline sdk/machinelearningservices/Azure.ResourceManager.MachineLearning/src/Generated/Models/Compute.cs:61]: `ResourceIdentifier`?

**Review by Yao725 — COMMENTED** [review]
Left a few comments, thanks!

File: sdk/machinelearningservices/Azure.ResourceManager.MachineLearning/src/Generated/CodeContainerDataData.cs [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+
+using System;
+using Azure.Core;
+using Azure.ResourceManager.MachineLearning.Models;
+using Azure.ResourceManager.Models;
+
+namespace Azure.ResourceManager.MachineLearning
+{
+    /// <summary> A class representing the CodeContainerData data model. </summary>
+    public partial class CodeContainerDataData : ResourceData
```
[the remark this answers is not in the document]
Comment (ArcturusZhang) [reply sdk/machinelearningservices/Azure.ResourceManager.MachineLearning/src/Generated/CodeContainerDataData.cs:16]: Well... this is caused by a `x-ms-client-name` in the swagger

File: sdk/machinelearningservices/Azure.ResourceManager.MachineLearning/src/Generated/Models/AutoMLVertical.cs (hunk 5 above)
Comment (ArcturusZhang) [reply sdk/machinelearningservices/Azure.ResourceManager.MachineLearning/src/Generated/Models/AutoMLVertical.cs:14]: Considering that ML is a two-letter acrynm, it should be ML

File: sdk/machinelearningservices/Azure.ResourceManager.MachineLearning/src/Generated/Models/AssignedUser.cs (hunk 4 above)
Comment (ArcturusZhang) [reply sdk/machinelearningservices/Azure.ResourceManager.MachineLearning/src/Generated/Models/AssignedUser.cs:13]: Probably. But I would like to defer this when we have more test cases. Now we have no test cases...

File: sdk/machinelearningservices/Azure.ResourceManager.MachineLearning/src/Generated/Models/BatchDeploymentDetails.cs [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+            Model = model;
+            OutputAction = outputAction;
+            OutputFileName = outputFileName;
+            ProvisioningState = provisioningState;
+            Resources = resources;
+            RetrySettings = retrySettings;
+        }
+
+        /// <summary> Compute target for batch inference operation. </summary>
+        public string Compute { get; set; }
```
[the remark this answers is not in the document]
Comment (ArcturusZhang) [reply sdk/machinelearningservices/Azure.ResourceManager.MachineLearning/src/Generated/Models/BatchDeploymentDetails.cs:69]: I am pretty sure this should be an Id of an ML compute instance, but I am not 100% sure. Let us defer this change when we have more test cases

File: sdk/machinelearningservices/Azure.ResourceManager.MachineLearning/src/Generated/Models/Compute.cs [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+using Azure.Core;
+
+namespace Azure.ResourceManager.MachineLearning.Models
+{
+    /// <summary> Machine Learning compute object. </summary>
+    public partial class Compute
```
[the remark this answers is not in the document]
Comment (ArcturusZhang) [reply sdk/machinelearningservices/Azure.ResourceManager.MachineLearning/src/Generated/Models/Compute.cs:15]: I already renamed this

**Review by ArcturusZhang — APPROVED** [review]

**Pull request merged**